incremental-font-transfer/src/font_patch.rs

```diff
@@ -74,31 +74,40 @@ impl From<SerializeErrorFlags> for PatchingError {
     }
 }
 
-impl From<ReadError> for PatchingError {
-    fn from(err: ReadError) -> Self {
-        PatchingError::FontParsingFailed(err)
-    }
-}
-
-impl From<DecodeError> for PatchingError {
-    fn from(decoding_error: DecodeError) -> Self {
+impl PatchingError {
+    pub(crate) fn from_patch_decode_error(decoding_error: DecodeError, uri: &str) -> Self {
         match decoding_error {
             DecodeError::InitFailure => {
-                PatchingError::InvalidPatch("Failure to init brotli encoder.")
+                PatchingError::InvalidPatch(format!("Failure to init brotli encoder. ({})", uri))
             }
-            DecodeError::InvalidStream => PatchingError::InvalidPatch("Malformed brotli stream."),
-            DecodeError::InvalidDictionary => PatchingError::InvalidPatch("Malformed dictionary."),
-            DecodeError::MaxSizeExceeded => PatchingError::InvalidPatch("Max size exceeded."),
-            DecodeError::ExcessInputData => {
-                PatchingError::InvalidPatch("Input brotli stream has excess bytes.")
+            DecodeError::InvalidStream(msg) => PatchingError::InvalidPatch(format!(
+                "Malformed brotli stream uri = {}, msg = {}",
+                uri, msg
+            )),
+            DecodeError::InvalidDictionary => {
+                PatchingError::InvalidPatch(format!("Malformed dictionary. ({})", uri))
             }
-            DecodeError::IoError(_) => {
-                PatchingError::InvalidPatch("IO error decoding input brotli stream.")
+            DecodeError::MaxSizeExceeded => {
+                PatchingError::InvalidPatch(format!("Max size exceeded. ({})", uri))
```

Comment:
I will reiterate my general distaste for errors with strings in them, on general principle 🤷